pennylane/devices/qubit/simulate.py

```diff
@@ -237,7 +257,297 @@ def simulate(
+) -> Result:
+    """Simulate a single quantum script with native mid-circuit measurements.
+
+    Args:
+        circuit (QuantumTape): The single circuit to simulate
+        rng (Union[None, int, array_like[int], SeedSequence, BitGenerator, Generator]): A
+            seed-like parameter matching that of ``seed`` for ``numpy.random.default_rng``.
+            If no value is provided, a default RNG will be used.
+        prng_key (Optional[jax.random.PRNGKey]): An optional ``jax.random.PRNGKey``. This is
+            the key to the JAX pseudo random number generator. If None, a random key will be
+            generated. Only for simulation using JAX.
+        debugger (_Debugger): The debugger to use
+        interface (str): The machine learning interface to create the initial state with
+        state_cache=None (Optional[dict]): A dictionary mapping the hash of a circuit to the pre-rotated state. Used to pass the state between forward passes and vjp calculations.
+
+    Returns:
+        tuple(TensorLike): The results of the simulation
+    """
+    if circuit.shots.has_partitioned_shots:
+        results = []
+        for s in circuit.shots:
+            aux_circuit = circuit.copy()
+            aux_circuit._shots = qml.measurements.Shots(s)
+            results.append(simulate(aux_circuit, rng, prng_key, debugger, interface))
+        return tuple(results)
+    aux_circuit = init_auxiliary_circuit(circuit)
+    all_shot_meas, mcm_values_dict = simulate_one_shot_native_mcm(
+        aux_circuit, rng, prng_key, debugger, interface
+    )
+    list_mcm_values_dict = [mcm_values_dict]
+    for _ in range(circuit.shots.total_shots - 1):
+        one_shot_meas, mcm_values_dict = simulate_one_shot_native_mcm(
+            aux_circuit, rng, prng_key, debugger, interface
+        )
+        all_shot_meas = accumulate_native_mcm(aux_circuit, all_shot_meas, one_shot_meas)
+        list_mcm_values_dict.append(mcm_values_dict)
+    return parse_native_mid_circuit_measurements(circuit, all_shot_meas, list_mcm_values_dict)
+
+
+def init_auxiliary_circuit(circuit: qml.tape.QuantumScript):
+    """Creates an auxiliary circuit to perform one-shot mid-circuit measurement calculations.
+
+    Measurements with non-trivial measurement values are removed from the script. VarianceMP
+    measurements are also replaced by SampleMP measurements, which are necessary to evaluate
+    the variance.
+
+    Args:
+        circuit (QuantumTape): The original QuantumScript
+
+    Returns:
+        QuantumScript: A copy of the circuit with modified measurements
+    """
+    aux_circuit = circuit.copy()
+    aux_circuit._shots = qml.measurements.Shots(1)
+    idx_sample = find_measurement_values(circuit)
+    for i in reversed(idx_sample):
+        aux_circuit._measurements.pop(i)
+    for i, m in enumerate(circuit.measurements):
+        if isinstance(m, VarianceMP) and m.mv is None:
+            aux_circuit._measurements[i] = SampleMP(obs=m.obs)
+    return aux_circuit
```

Comment:
```suggestion
    idx_sample = find_measurement_values(circuit)
    new_measurements = []
    for i, m in enumerate(circuit.measurements):
        if i not in idx_sample:
            if isinstance(m, VarianceMP) and m.mv is None:
                new_measurements.append(SampleMP(obs=m.obs))
            else:
                new_measurements.append(m)
    return qml.tape.QuantumScript(circuit.operations, new_measurements,
        shots=1,
        trainable_params=circuit.trainable_params)
```

Should be equivalent.  This way we don't have to worry about properties calculated on initialization getting out of date.